### ISSUE TITLE
Remove implicit usings from newly generated classes

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#59922]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Remove-implicit-usings-from-newly-genera/1681796)._

---
If .Net 6.0 supports implicit usings and implict usings is enabled on the project, don't generate classes with a a bunch of explicit system usings that would otherwise not be required.

---
#### Feedback Bot on 3/2/2022, 07:06 PM: 

Thank you for taking the time to provide your suggestion. We will do some preliminary checks to make sure we can proceed further. You will hear from us in about a week on our next steps.

**Comment by davidwengier** [comment]
Possibly a duplicate of https://github.com/dotnet/roslyn/issues/47506 because in general we don't want to remove unused usings in new files.

We could potentially introduce a new refactoring to remove usings that are duplicated by global usings, but that is blocked by https://github.com/dotnet/roslyn/issues/59088

**Comment by Arithmomaniac** [comment]
@davidwengier it looks like #59088 was completed a while ago now?

**Comment by davidwengier** [comment]
It was. The word "potentially" is doing a lot of heavy lifting in my above comment :)

Also worth noting, this would not currently be technically possible to do for new projects, only new classes within existing projects. Whether that affects it's priority or not is another question.

**Comment by TXRock** [comment]
Current behavior for Visual Studio 17.7.5:

- New library project with auto-created Class1.cs does not contain any of the implicit usings.
- Creating new classes in an existing project contains those 5 implicit usings. Which plenty of developers find unnecessary.

**Comment by radderz** [comment]
This is definitely something that is frustrating to do when building a bunch of model files. It's not important but it is quite annoying.

**Comment by TXRock** [comment]
> Current behavior for Visual Studio 17.7.5:
> 
> * New library project with auto-generated Class1.cs does not contain any of the implicit usings.
> * Creating new classes in an existing project contains those 5 implicit usings. Which plenty of developers find unnecessary.

Still, one can always customize the `C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\ItemTemplates\CSharp\Code\2052\Class\Class.cs` file for a template one particularly aims for when creating a new class. Be sure to backup the original file in case of any mistakes made to the file.

e.g.

```csharp
namespace $rootnamespace$
{
    public class $safeitemrootname$
    {
    }
}
```

**Comment by radderz** [comment]
Seems hacky and prone to being a different directory after updates?

For instance I think mine is located here, 
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\ItemTemplates\CSharp\Code\1033\Class\Class.cs

**Comment by TXRock** [comment]
> Seems hacky and prone to being a different directory after updates?
> 
> For instance I think mine is located here, C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\ItemTemplates\CSharp\Code\1033\Class\Class.cs

You are correct. I cannot help with what Microsoft would choose to do. Just giving an option in case someone really needs to do so.

**Comment by radderz** [comment]
Ī eddited that file to be:

`$if$ ($targetframeworkversion$ < 7.0)using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;
$endif$namespace $rootnamespace$;

class $safeitemrootname$
{
}
`

incase anyone else is interested.

I don't ever do anything with .NET 3.5 so the existing template was pretty redundant for anything other than pre .NET 7 when the implicit usings was brought in.

**Comment by CyrusNajmabadi** [comment]
> Seems hacky and prone to being a different directory after updates?
> 
> For instance I think mine is located here, 
> C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\ItemTemplates\CSharp\Code\1033\Class\Class.cs

Definitely not a hack. This is the expected way to change these templates.
